### ISSUE TITLE
logic primacy and/or logic nesting

### ISSUE THREAD

**Issue by kahn10** [rndusr/stig#181]
I'm trying to understand exactly how stig handles complex logic groupings.  Do 'and' statements group 'or' statements or vice versa? is hierarchical logic possible?

Example:
```
stig ls 'foo|bar&blah'
```
Does this produce all items which have 'foo' or 'bar', all of which contain 'blah', or does it deliver all 'foo' along with items which contain both 'bar' and 'blah'? It looks like it's the latter., but I want to confirm I'm interpreting that correctly. Would this produce the same list as:
```
stig ls 'bar&blah|foo'
```
Is it possible to use brackets or braces to intentionally nest logic statements?

**Comment by kahn10** [comment]
Further testing has made me conclude that 'or' statements group 'and' statements. I can work with that, but I wonder if explicit nesting could be implemented.

**Comment by rndusr** [comment]
Yes, IIRC, first all AND groups are evaluated and the results are then ORed.

Explicit grouping would definitely be nice, but it's not a very pressing issue
in my opinion.

PRs are welcome.
